plugins/repository-s3/src/test/java/org/elasticsearch/repositories/s3/S3BlobContainerRetriesTests.java

```diff
@@ -192,21 +194,34 @@ public void testWriteBlobWithRetries() throws Exception {
         final int maxRetries = randomInt(5);
         final CountDown countDown = new CountDown(maxRetries + 1);
 
-        final byte[] bytes = randomByteArrayOfLength(randomIntBetween(1, 512));
+        final byte[] bytes = randomByteArrayOfLength(randomIntBetween(1, frequently() ? 512 : 1 << 20)); // rarely up to 1mb
+        final long expectedChecksum = checksum(new ByteArrayInputStream(bytes), bytes.length);
```

Comment:
Why do we need the checksum? :D I'm not sure I understand why it's better here it just seems more confusing :)

Reply:
In the case of a larger blob size is picked up (say, 1mb) I thought that using a checksum instead of a deep array comparison was more efficient.

Reply:
Idk, either way you have to read all the bytes once for each array to checksum it and comparing byte arrays from heap memory is incredibly cheap anyway? :) Not sure this is worth the added complexity.

Reply:
Ok, then I'll remove the checksum. Thanks